src/main/java/org/jabref/logic/openoffice/OpenOfficeFileSearch.java

```diff
@@ -1,121 +1,80 @@
 package org.jabref.logic.openoffice;
 
 import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
-import java.util.Optional;
-
-public class OpenOfficeFileSearch {
-
-    private boolean fileSearchCanceled;
+import java.util.Locale;
+import java.util.stream.Collectors;
 
+import org.jabref.logic.util.OS;
+import org.jabref.logic.util.io.FileUtil;
 
+public class OpenOfficeFileSearch {
     /**
-     * Search for Program files directory.
-     * @return the File pointing to the Program files directory, or null if not found.
-     *   Since we are not including a library for Windows integration, this method can't
-     *   find the Program files dir in localized Windows installations.
+     * Detects existing installation of OpenOffice and LibreOffice.
+     *
+     * @return a list of detected installation paths
      */
-    public List<File> findWindowsProgramFilesDir() {
-        List<String> sourceList = new ArrayList<>();
-        List<File> dirList = new ArrayList<>();
-
-        // Check default 64-bits program directory
-        String progFiles = System.getenv("ProgramFiles");
-        if (progFiles != null) {
-            sourceList.add(progFiles);
+    public static List<Path> detectInstallations() {
+        if (OS.WINDOWS) {
+            List<Path> programDirs = findWindowsOpenOfficeDirs();
+            return programDirs.stream().filter(dir -> FileUtil.find(OpenOfficePreferences.WINDOWS_EXECUTABLE, dir).isPresent()).collect(Collectors.toList());
+        } else if (OS.OS_X) {
+            List<Path> programDirs = findOSXOpenOfficeDirs();
+            return programDirs.stream().filter(dir -> FileUtil.find(OpenOfficePreferences.OSX_EXECUTABLE, dir).isPresent()).collect(Collectors.toList());
+        } else if (OS.LINUX) {
+            List<Path> programDirs = findLinuxOpenOfficeDirs();
+            return programDirs.stream().filter(dir -> FileUtil.find(OpenOfficePreferences.LINUX_EXECUTABLE, dir).isPresent()).collect(Collectors.toList());
         }
+        return new ArrayList<>(0);
+    }
 
-        // Check default 64-bits program directory
-        progFiles = System.getenv("ProgramFiles(x86)");
-        if (progFiles != null) {
-            sourceList.add(progFiles);
-        }
+    private static List<Path> findOpenOfficeDirectories(List<Path> programDirectories) {
+        List<Path> result = new ArrayList<>();
 
-        for (String rootPath : sourceList) {
-            File root = new File(rootPath);
-            File[] dirs = root.listFiles(File::isDirectory);
-            if (dirs != null) {
-                for (File dir : dirs) {
-                    if (dir.getPath().contains("OpenOffice") || dir.getPath().contains("LibreOffice")) {
-                        dirList.add(dir);
+        for (Path programDir : programDirectories) {
+            File[] subDirs = programDir.toFile().listFiles(File::isDirectory);
```

Comment:
I think Files.find or Files.walk can be used here as well, because a path can also be a directory